clients/admin-ui/src/features/system/form.ts

```diff
@@ -105,6 +105,7 @@ export const transformFormValuesToSystem = (
         ...payload.meta.vendor,
         id: formValues.meta.vendor.id,
       };
+      payload.vendor_id = formValues.meta.vendor.id;
```

Comment:
This can be tested by looking in the network tab when you update a system using dictionary data.  The system response should now have a vendor_id populated!